### PR TITLE
Fixture in `test_examples.py` to clean up generated data files

### LINKED ISSUE #1764: Avoiding pytest suite side-effects (dataset saving)
Running the test suite creates the following zarr files:

```
./Peninsula.zarr
./MIT_particles_scipy.zarr
./nemo_particles.zarr
./DecayingMovingParticle.zarr
./StommelParticle.zarr
./EddyParticle.zarr
./RadialParticle.zarr
./MIT_particles_jit.zarr
```

cluttering the working directory. Changing the tests to clean up these files after the tests have run (e.g., by using temporary files) would improve the quality of the test suite.

### PR DESCRIPTION
**Pull request by VeckoTheGecko**
Fixes #1764

No more side effects from running the test suite 🎉

### CODE REVIEW

**Comment by VeckoTheGecko** [comment]
Not 100% sure if the buffer is necessary. I get a "PermissionError" on a file one time but that was a once off.

File: tests/test_examples.py [hunk 1]
```diff
@@ -8,6 +11,33 @@
...
+            continue
+        if not (fname.endswith(".nc") or fname.endswith(".zarr")):
+            continue
+
+        path = Path(fname)
+        if path.is_dir():
+            shutil.rmtree(path)
+        else:
+            path.unlink()
+        print(f"Removed {path}")
```
Comment (erikvansebille) [inline tests/test_examples.py:38]: Is this print really necessary?

**Review by erikvansebille — COMMENTED** [review]
Nice solution! Will this now also work on other files in the test_suite (e.g. test_particlfile.py, or many of the notebooks in docs/examples/*.ipynb that generate output zarr files)? Or does the fixture only work in the file (test_examples.py) in which it is created? Would be nice to have this for the entire test_suite?

**Comment by VeckoTheGecko** [comment]
The fixture is scoped to `test_examples.py` as it is unique to the `test_example_script()` function and how it invokes the scripts as these scripts aren't normal pytest tests. Normal Pytest tests should use the `tmpdir` fixture to handle cleanup.

Running `pytest` (no arguments - this is the unit tests - which runs the functions named `test_*` in `tests` and `docs/examples` as well as runs the `__main__` scipts in `docs/examples/example_*.py`) now doesn't generate any loose data files.

Running the tutorial notebooks will still generate loose data files. I don't think its worth cleaning these up automatically as they're saved to `docs/examples` instead of the project root so are out of the way, and also the notebooks aren't normally run via pytest anyway. Devs can do `find docs/examples -name '*.zarr' | xargs rm -r` to remove this generated data.

File: tests/test_examples.py (hunk 1 above)
Comment (VeckoTheGecko) [reply tests/test_examples.py:38]: Not particularly, but I think its nice for debugging. It only displays when running Pytest with `--capture=no`.

**Review by erikvansebille — APPROVED** [review]

**Pull request merged**